### PR TITLE
[EOC-52] error handling for API request

### PR DESCRIPTION
**Pull request by alemikolo**

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -1,3 +1,3 @@
 const app = require('./app');
 
-app.listen(8080, () => console.info('Listening on port 8080!'));
```
Comment (marekrozmus) [inline server/index.js]: please revert and add lint ignore next line

File: client/src/modules/shopping-list/components/ProductsList/actions.js [hunk 2]
```diff
@@ -1,32 +1,66 @@
...
-    .catch(err => console.error(err));
+    .then(product =>
+      setTimeout(() => dispatch(toggleProductSuccess(product)), 600)
+    )
+    .catch(err => {
+      dispatch(toggleProductFailure());
+      createNotificationWithTimeout(
+        dispatch,
+        NotificationType.ERROR,
+        "Oops, we're sorry, setting item as done failed..."
```
Comment (adamklepacz) [inline client/src/modules/shopping-list/components/ProductsList/actions.js]: I'm not sure if this message will be correct while toggling back from history list to list

File: client/src/modules/shopping-list/model/actions.js [hunk 3]
```diff
@@ -1,45 +1,89 @@
...
 import { ShoppingListActionTypes } from './actionTypes';
-
+import { MessageType as NotificationType } from 'common/constants/enums';
+import { createNotificationWithTimeout } from 'modules/notification/model/actions';
 // Action creators
-export const fetchProductsError = err => ({
+const fetchProductsFailure = errMessage => ({
   type: ShoppingListActionTypes.FETCH_PRODUCTS_FAILURE,
-  err
+  errMessage
```
Comment (adamklepacz) [inline client/src/modules/shopping-list/model/actions.js]: same about here, consider using payload

File: client/src/modules/shopping-list/model/actions.js [hunk 4]
```diff
@@ -1,45 +1,89 @@
...
 });
-export const recieveProducts = json => ({
-  type: ShoppingListActionTypes.FETCH_PRODUCTS_REQUEST,
+const fetchProductSuccess = json => ({
+  type: ShoppingListActionTypes.FETCH_PRODUCTS_SUCCESS,
   products: json
```
Comment (adamklepacz) [inline client/src/modules/shopping-list/model/actions.js]: leave products here just for now, and in other new created actions - make payload

File: client/src/modules/cohort/model/selectors.js [hunk 5]
```diff
@@ -1 +1,3 @@
-export const getCohorts = state => state.cohorts;
+export const getCohorts = state => state.cohorts.data;
+export const getCohortsError = state => state.cohorts.errorMessage;
+export const getIsFetchingCohorts = state => state.cohorts.isFetching;
```
[the remark this answers is not in the document]
Comment (alemikolo) [reply client/src/modules/cohort/model/selectors.js]: Later FetchingStatus could have few values. Now, IsFetching is a boolean. getIsFetchingCohorts
is a more appropriate name for variable of type boolean

**Pull request merged**